File: javascript/algorithm/search/sequence.js
/**
 * 线性查找
 */

function sequenceSearch(nums, num) {
  if (nums == null) {
    return -1;
  }
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] == num) {
      return i;
    }
  }
  return -1;
  }
  
  function main() {
    let value = 8;
    let nums = [2, 5, 1, 7, 8, 16];
    let index = sequenceSearch(nums, value);
    if (index != -1) {
      console.log(value + " 在 nums 中的下标为: " + index);
    }
  }
  
  main();
  

File: python/algorithm/search/sequence.py
"""
线性查找
"""


def sequence_search(nums: list[int] | None, num: int) -> int:
    if nums is None:
        return -1
    for i, value in enumerate(nums):
        if value == num:
            return i
    return -1


def main() -> None:
    value = 8
    nums = [2, 5, 1, 7, 8, 16]
    index = sequence_search(nums, value)
    if index != -1:
        print(f"{value} 在 nums 中的下标为: {index}")


if __name__ == "__main__":
    main()
